util/parsec_result.py

```diff
@@ -0,0 +1,95 @@
+"""Represents the result of running ParseC over C source code."""
+
+from __future__ import annotations
+
+import json
+import os
+import subprocess
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Any
+
+import networkx as nx
+
+from util.parsec_function import ParsecFunction
+
+
+@dataclass
+class ParsecResult:
+    """Represents the top-level result obtained by running `parsec` on a C file.
+
+    ParseC is a LLVM/Clang-based tool to parse a C program (hence the name).
+    It extracts functions, structures, etc. along with their inter-dependencies.
+    """
+
+    enums: list[Any] = field(default_factory=list)
+    files: list[str] = field(default_factory=list)
+    functions: dict[str, ParsecFunction] = field(default_factory=dict)
+
+    def __init__(self, file_path: Path):
```

Comment:
It is best to document, especially with a generic name like `file_path`:  it is a JSON file written by ParseC.